Common subdirectories: cricket-0.70-original/doc and cricket-0.70-modified/doc
diff -u cricket-0.70-original/grapher.cgi cricket-0.70-modified/grapher.cgi
--- cricket-0.70-original/grapher.cgi	Tue Aug 10 04:54:17 1999
+++ cricket-0.70-modified/grapher.cgi	Thu Jan  6 10:51:28 2000
@@ -175,6 +175,8 @@
 			# make the instance selection widget...
 			htmlHeader($name, $targRef, "Instance selection for $tname");
 
+			print htmlCurrentPath($ct, $name);
+
 			print "There are multiple instances for this target. Please";
 			print " choose one:<p>\n";
 
@@ -329,6 +331,8 @@
 
 			htmlHeader($name, $targRef, $title);
 
+			print htmlCurrentPath($ct, $name);
+
 			print "<table width=100% cellpadding=5 padding=3 border>\n";
 			print "<tr><td width=70%>\n";
 
@@ -582,6 +586,8 @@
 			$orderb <=> $ordera || $a cmp $b;
 		} @targets;
 		
+		print htmlCurrentPath($ct, $name);
+
 		if ($#targets+1 > 0) {
             my($doDesc) = 1;
             if ($targs->{$targets[0]}->{'disable-short-desc'}) {
@@ -1781,6 +1787,37 @@
 		$i++;
 	}
 	return @links;
+}
+
+sub htmlCurrentPath {
+	my($ct, $target) = @_;
+	my($html);
+
+	if ($target !~ /^\s*\/\s*$/) {
+		$html = "<font size=12><b>Current path:</b></font>\n";
+		$html .= htmlCurrentPathLinks($ct, $target);
+		$html .= "<p>\n";
+	}
+
+	return $html;
+}
+ 
+sub htmlCurrentPathLinks {
+	my($ct, $target) = @_;
+	my($html);
+ 
+	my($path) = "/";
+	foreach $p (split(/\//, $target)) {
+		my($lQ) = new CGI;
+		$path .= "$p";
+		$lQ->delete_all() unless ($ct->isLeaf($path));
+		$lQ->param('target', $path);
+		$p .= "/" unless ($p =~ /\/$/) || ($ct->isLeaf($path));
+		$path .= "/" unless $path =~ /\/$/;
+		$html .= "&nbsp;&nbsp;&nbsp;<a href=\"" . $lQ->self_url() . "\">" . $p . "</a>\n";
+	}
+ 
+	return $html;
 }
 
 sub generateImageName {
Common subdirectories: cricket-0.70-original/images and cricket-0.70-modified/images
Common subdirectories: cricket-0.70-original/lib and cricket-0.70-modified/lib
Common subdirectories: cricket-0.70-original/sample-config and cricket-0.70-modified/sample-config
Common subdirectories: cricket-0.70-original/util and cricket-0.70-modified/util

